src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -183,46 +182,55 @@ private void UpdateSize()
             if (extraField.Tag != TagConstant)
                 return false;
 
-            // this pattern needed because nested using blocks trigger CA2202
-            MemoryStream? ms = null;
-            try
-            {
-                ms = new MemoryStream(extraField.Data);
-                using (BinaryReader reader = new BinaryReader(ms))
-                {
-                    ms = null;
+            zip64Block._size = extraField.Size;
 
-                    zip64Block._size = extraField.Size;
+            using (MemoryStream ms = new MemoryStream(extraField.Data))
+            using (BinaryReader reader = new BinaryReader(ms))
+            {
+                // The spec section 4.5.3:
+                //      The order of the fields in the zip64 extended
+                //      information record is fixed, but the fields MUST
+                //      only appear if the corresponding Local or Central
+                //      directory record field is set to 0xFFFF or 0xFFFFFFFF.
+                // However tools commonly write the fields anyway; the prevailing convention
+                // is to respect the size, but only actually use the values if their 32 bit
+                // values were all 0xFF.
+
+                if (ms.Position + 8 > extraField.Size)
```

Comment:
```suggestion
                if (ms.Position > extraField.Size - 8)
```
Paranoia of overflow 😉